staging/src/k8s.io/client-go/transport/cache_go118.go

```diff
@@ -0,0 +1,22 @@
+//go:build go1.18
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transport
+
+// isComparable asserts at compile time that tlsCacheKey is comparable in a way that will never panic at runtime.
+func isComparable[T comparable]() { isComparable[tlsCacheKey]() }
```

Comment:
could this be in a _test file and be equally effective?

also, is there a way to construct this that doesn't make a function that infinitely recursively calls itself, like
```go
func isComparable[T comparable]() bool { return true }

var _ = isComparable[tlsCacheKey]()
```

Reply:
> could this be in a _test file and be equally effective?

We don't generally put interface assertions in test files because you only get a failure if you actually compile/run the test.  Having it in the regular file make its impossible to miss.

> also, is there a way to construct this that doesn't make a function that infinitely recursively calls itself

Yeah, that is basically what I had originally and I got too clever with my desire to fit in in one line :)  (also our CI linter doesn't like that the function is unused so I have to fix it anyway)

Reply:
> We don't generally put interface assertions in test files

`git grep "var _ = " | grep _test.go`  disagrees :)

> because you only get a failure if you actually compile/run the test.

Which we do, right?

Reply:
> > We don't generally put interface assertions in test files
> 
> `git grep "var _ = " | grep _test.go` disagrees :)

Hence I used generally.  When we really want to make sure stuff does not drift we do it in the regular code like:

https://github.com/kubernetes/kubernetes/blob/664f0f567777c91dfc3b2cf694fbc1af2666f853/pkg/registry/core/service/storage/storage.go#L70-L76

> > because you only get a failure if you actually compile/run the test.
> 
> Which we do, right?

Why push an assertion down to a later time?